### DISCUSSION TITLE
Unique select option for field arrays?

### DISCUSSION THREAD

**Discussion by kujohn** [react-hook-form/react-hook-form#8395] [Q&A]
In the field array, one of the input is a select dropdown that only shows unselected options. Is there a pattern to build and eliminate  already selected fields? Example:

<img width="430" alt="Screen Shot 2022-05-23 at 5 01 37 PM" src="https://user-images.githubusercontent.com/1356936/169922882-05a7fc09-6698-4317-9015-4f0c9aac4ff5.png">

**Comment by Moshyfawn** [comment]
You can simply `watch` your field's value and `.filter` your options against the already selected values.

**Reply by kujohn** [reply]
I've tried `.filter` on the list but the already selected option gets taken out. I guess I could try making exception if current input field is already selected.

**Reply by Moshyfawn** [reply]
I was thinking of something along the lines of:
```js
  const getPrevSelectedOptions = (fieldIndex) =>
    getValues("fields")
      .slice(0, fieldIndex)
      .flatMap((field) => field.select);
```
```jsx
<select>
  {options
    .filter(
      (option) =>
        !getPrevSelectedOptions(index).some(
          (prevSelectedOption) => prevSelectedOption === option
        )
    )
    .map((option) => (
      <option value={option}>
        {option}
      </option>
    ))}
</select>
```
See [this codesandbox](https://codesandbox.io/s/gifted-benji-jpjwgy?file=%2Fsrc%2FApp.tsx)
